azurecompute/src/main/java/org/jclouds/azurecompute/domain/VMImageParams.java

```diff
@@ -0,0 +1,247 @@
+
+      public Builder description(String description) {
+         this.description = description;
+         return this;
+      }
+
+      public Builder osDiskConfiguration(OSDiskConfigurationParams osDiskConfig) {
+         this.osDiskConfiguration = osDiskConfig;
+         return this;
+      }
+
+      public Builder language(String language) {
+         this.language = language;
+         return this;
+      }
+
+      public Builder imageFamily(String imageFamily) {
+         this.imageFamily = imageFamily;
+         return this;
+      }
+
+      public Builder recommendedVMSize(RoleSize.Type recommendedRoleSize) {
+         this.recommendedVMSize = recommendedRoleSize;
+         return this;
+      }
+
+      public Builder showGui(Boolean showGui) {
+         this.showGui = showGui;
+         return this;
+      }
+
+      public Builder eula(String eula) {
+         this.eula = eula;
+         return this;
+      }
+
+      public Builder iconUri(URI iconUri) {
+         this.iconUri = iconUri;
+         return this;
+      }
+
+      public Builder smallIconUri(URI smallIconUri) {
+         this.smallIconUri = smallIconUri;
+         return this;
+      }
+
+      public Builder privacyUri(URI privacyUri) {
+         this.privacyUri = smallIconUri;
+         return this;
+      }
+
+      public Builder dataDiskConfiguration(DataVirtualHardDisk dataDiskConfiguration) {
+         this.dataDiskConfiguration.add(dataDiskConfiguration);
+         return this;
+      }
+
+      public Builder dataDiskConfigurations(Collection<DataVirtualHardDisk> dataDiskConfiguration) {
+         this.dataDiskConfiguration.addAll(dataDiskConfiguration);
+         return this;
+      }
+
+      public VMImageParams build() {
+         return VMImageParams
+               .create(name, label, description, osDiskConfiguration, dataDiskConfiguration, language,
+                     imageFamily, recommendedVMSize, eula, iconUri, smallIconUri, privacyUri,
+                     showGui);
+      }
+
+      public Builder fromVMImageParams(VMImageParams in) {
+         return name(in.name()).label(in.label()).description(in.description())
+               .osDiskConfiguration(in.osDiskConfiguration()).dataDiskConfigurations(in.dataDiskConfiguration())
+               .language(in.language()).imageFamily(in.imageFamily()).recommendedVMSize(in.recommendedVMSize())
+               .eula(in.eula()).iconUri(in.iconUri()).smallIconUri(in.smallIconUri()).privacyUri(in.privacyUri());
+      }
+
+   }
+
+   public static VMImageParams create(String name, String label, String description,
+         OSDiskConfigurationParams osDiskConfiguration, List<DataVirtualHardDisk> dataDiskConfiguration,
+         String language, String imageFamily, RoleSize.Type recommendedVMSize, String eula, URI iconUri,
+         URI smallIconUri, URI privacyUri, Boolean showGui) {
+      return new AutoValue_VMImageParams(name, label, description, osDiskConfiguration, dataDiskConfiguration,
```

Comment:
for lists (or any other collections) in AutoValue classe, please use ImmutableList.copyOf(dataDiskConfiguration) as recommended by auto/value doc